### PR TITLE
RSVP + Meta for QA feedback

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#38995](https://central.tri.be/issues/38995) [C#43942](https://central.tri.be/issues/43942)
_Rel:_ #166 moderntribe/event-tickets-plus#153

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Each line in an array declaration must end in a comma | [Tickets_View.php:262](https://github.com/moderntribe/event-tickets/blob/96ba778bdabcfa46dff2b2dc17a35b2e616f04bd/src/Tribe/Tickets_View.php#L262)

> via jenkins/codesniffer-phpcs

File: src/Tribe/RSVP.php [hunk 1]
```diff
@@ -674,6 +674,9 @@ protected function get_tickets( $event_id ) {
 		return $tickets;
 	}
 
+
+	private $done = false;
```
Comment (borkweb) [inline src/Tribe/RSVP.php]: Can you move this up to the top of the class file?

File: src/Tribe/Tickets_View.php [hunk 2]
```diff
@@ -163,8 +195,10 @@ public function update_tickets() {
 	 * @return void
 	 */
 	public function authorization_redirect() {
+		global $wp_query;
+
 		// When it's not our query we don't care
-		if ( ! tribe_is_event_query() ) {
+		if ( empty( $wp_query->tribe_is_event_query ) ) {
```
Comment (borkweb) [inline src/Tribe/Tickets_View.php]: Why the change here? It feels like we'd be better served to continue using `tribe_is_event_query()` so the value can be filtered.

File: src/Tribe/Tickets_View.php [hunk 3]
```diff
@@ -241,8 +290,10 @@ public function intercept_content( $content ) {
 	 * @return string           The correct File path for the tickets endpoint
 	 */
 	public function intercept_template( $old_file, $template ) {
+		global $wp_query;
+
 		// When it's not our query we don't care
-		if ( ! tribe_is_event_query() ) {
+		if ( empty( $wp_query->tribe_is_event_query ) ) {
```
Comment (borkweb) [inline src/Tribe/Tickets_View.php]: Same question about the switch away from [`tribe_is_event_query()`](https://github.com/moderntribe/the-events-calendar/blob/57c50037b6e91535b5108dcc9f1736a54d9679dd/src/functions/template-tags/query.php#L51) as asked above.

File: src/Tribe/Tickets_View.php (hunk 2 above)
Comment (bordoni) [inline src/Tribe/Tickets_View.php]: Doesn't exist on Tickets, only on Core: `tribe_is_event_query()`

File: src/Tribe/Tickets_View.php (hunk 3 above)
Comment (borkweb) [inline src/Tribe/Tickets_View.php]: Same question about the switch away from [`tribe_is_event_query()`](https://github.com/moderntribe/the-events-calendar/blob/57c50037b6e91535b5108dcc9f1736a54d9679dd/src/functions/template-tags/query.php#L51) as asked above.

Comment (bordoni) [inline src/Tribe/Tickets_View.php]: Same Answer as the Above 😄

File: src/Tribe/Tickets_View.php (hunk 2 above)
Comment (borkweb) [inline src/Tribe/Tickets_View.php]: Hrm. Well then. Perhaps we should add a function to tribe-common that TEC can hook to so that event-tickets doesn't need to be aware of TEC stuff? Maybe a function called `tribe_is_modified_query()` that has a `tribe_is_modified_query` filter. Then core can use `tribe_is_event_query()` to impact that filtered value in some way?

Comment (bordoni) [inline src/Tribe/Tickets_View.php]: Im not really sure... Im thinking if there is another way to go about it. The `tribe_is_modified_query` is a little bit weird for me, has too much of wide usage. Should we go with adding the filter here on Tickets? So that if it's filtered it still applies?

Comment (borkweb) [inline src/Tribe/Tickets_View.php]: I like it.

**Comment by bordoni** [comment]
Ready for another look.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_View.php:201](https://github.com/moderntribe/event-tickets/blob/9ca89b5182ecd75609aea9b9cdf3755ba3c74e2b/src/Tribe/Tickets_View.php#L201)
:warning: | Comment refers to a TODO task | [Tickets_View.php:301](https://github.com/moderntribe/event-tickets/blob/9ca89b5182ecd75609aea9b9cdf3755ba3c74e2b/src/Tribe/Tickets_View.php#L301)
:warning: | Comment refers to a TODO task | [Tickets_View.php:364](https://github.com/moderntribe/event-tickets/blob/9ca89b5182ecd75609aea9b9cdf3755ba3c74e2b/src/Tribe/Tickets_View.php#L364)

> via jenkins/codesniffer-phpcs

**Comment by borkweb** [comment]
Cool. This looks 🚂 :dash:
